### PR TITLE
fix: types and managing of stream data

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Updated the types to follow what was explained in the DR.
- Simplified state management.
- Moved away from using a `useEffect` for triggering the prompt instead of executing right there and then.
- Introduced a SearchContainer to be used for mobile app's modal version.
- Updated the stream types handlers.
- Should be ready for what is expected in the DR.

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1653 #done

### CODE REVIEW

File: packages/shared/src/hooks/useChat.ts
Comment (capJavert) [inline packages/shared/src/hooks/useChat.ts:140]: My only concern with this approach If hook unmounts the stream will be left open.

Comment (capJavert) [reply packages/shared/src/hooks/useChat.ts:140]: We can probably clean it in empty useEffect with only cleanup function.

Comment (sshanzel) [reply packages/shared/src/hooks/useChat.ts:140]: Oh, right. Yes, let's open a `useEffect` to clean it up.

File: packages/shared/src/components/search/SearchContainer.tsx
Comment (rebelchris) [inline packages/shared/src/components/search/SearchContainer.tsx:16]: Is the grid necessary here?
If not prefer flex in case things dynamic render.

**Review by rebelchris — APPROVED** [review]
Nothing that seems off to me.
But let's wait for @capJavert response before merging.

Just wondering does this already cover the loader changes based on state or will it be seperate?

**Review by capJavert — APPROVED** [review]
Looks good, I guess any changes needed can be done when we actually have real implementation 🚀

File: packages/shared/src/components/search/SearchContainer.tsx
Comment (sshanzel) [reply packages/shared/src/components/search/SearchContainer.tsx:16]: Oh, this was the existing classes when I moved this container. Let me check if it is necessary.

**Comment by sshanzel** [comment]
> Looks good, I guess any changes needed can be done when we actually have real implementation 🚀

I agree. Things might still change once the actual integration becomes available. Let's see how it goes 🚢

File: packages/shared/src/components/search/SearchContainer.tsx
Comment (sshanzel) [reply packages/shared/src/components/search/SearchContainer.tsx]: Seems okay with flex 🥳

**Pull request merged**
